OpenApiAnnotationScanner throws StringIndexOutOfBoundsException on primitive array types

Given a JAX-RS response class that provides data as an array of primitives, an exception is always thrown.

```java
@Schema(name = "MyResponse", description = "the REST response class"
public class MyResponse {
    @Schema(required = true, description = "a packed data array")
    private double[] data;

    // ... getters and setters removed
}
```

Then a given a simple jax-rs service
```java
@Path("/v1")
public class MyResource {
    @GET
    @Operation(summary = "get a response")
    @APIResponses({@APIResponse(responseCode="200", 
                                content = @Content(mediaType = "application/json", 
                                                                   schema = @Schema(implementationClass = MyResponse.class)))})
    public MyResponse getResponse() {
        return new MyResponse();
    }
}
```

This returns the following stacktrace
```
15:46:27,679 ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build threw an exception: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 6
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:141)
        at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:171)
        at io.quarkus.dev.DevModeMain.main(DevModeMain.java:89)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor#build threw an exception: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 6
        at io.quarkus.builder.Execution.run(Execution.java:108)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:121)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:115)
        at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:107)
        ... 2 more
Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 6
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
        at java.base/java.lang.String.substring(String.java:1874)
        at io.smallrye.openapi.runtime.scanner.FilteredIndexView.accepts(FilteredIndexView.java:67)
        at io.smallrye.openapi.runtime.scanner.FilteredIndexView.getClassByName(FilteredIndexView.java:103)
        at org.jboss.jandex.CompositeIndex.getClassByName(CompositeIndex.java:190)
        at io.smallrye.openapi.runtime.scanner.dataobject.AugmentedIndexView.getClass(AugmentedIndexView.java:43)
        at io.smallrye.openapi.runtime.scanner.dataobject.AugmentedIndexView.containsClass(AugmentedIndexView.java:47)
        at io.smallrye.openapi.runtime.scanner.dataobject.IgnoreResolver$JsonIgnoreTypeHandler.shouldIgnore(IgnoreResolver.java:226)
        at io.smallrye.openapi.runtime.scanner.dataobject.IgnoreResolver.isIgnore(IgnoreResolver.java:72)
        at io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.depthFirstGraphSearch(OpenApiDataObjectScanner.java:223)
        at io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.process(OpenApiDataObjectScanner.java:190)
        at io.smallrye.openapi.runtime.scanner.OpenApiDataObjectScanner.process(OpenApiDataObjectScanner.java:141)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.introspectClassToSchema(OpenApiAnnotationScanner.java:1820)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.readClassSchema(OpenApiAnnotationScanner.java:1803)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.readSchema(OpenApiAnnotationScanner.java:1757)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.readSchema(OpenApiAnnotationScanner.java:1701)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.readMediaType(OpenApiAnnotationScanner.java:1535)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.readContent(OpenApiAnnotationScanner.java:1499)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.readResponse(OpenApiAnnotationScanner.java:1665)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.processJaxRsMethod(OpenApiAnnotationScanner.java:622)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.processJaxRsResourceClass(OpenApiAnnotationScanner.java:362)
        at io.smallrye.openapi.runtime.scanner.OpenApiAnnotationScanner.scan(OpenApiAnnotationScanner.java:200)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.generateAnnotationModel(SmallRyeOpenApiProcessor.java:265)
        at io.quarkus.smallrye.openapi.deployment.SmallRyeOpenApiProcessor.build(SmallRyeOpenApiProcessor.java:217)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:768)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:415)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1535)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1426)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

The exception appears to come from [FilteredIndexView line 70](https://github.com/smallrye/smallrye-open-api/blob/master/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/FilteredIndexView.java#L70) because an array isn't going to have a '.' in the name.

Changing `double[]` to a `List<Double>` resolves the issue.  However that doesn't seem like the correct solution.

--- a/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/FilteredIndexView.java
+++ b/implementation/src/main/java/io/smallrye/openapi/runtime/scanner/FilteredIndexView.java
@@ -67,7 +67,8 @@ public class FilteredIndexView implements IndexView {
      */
     private boolean accepts(DotName className) {
         String fqcn = className.toString();
-        String packageName = fqcn.substring(0, fqcn.lastIndexOf('.'));
+        int index = fqcn.lastIndexOf('.');
+        String packageName = index > -1 ? fqcn.substring(0, index) : "";
         boolean accept;
         // Includes
         if (scanClasses.isEmpty() && scanPackages.isEmpty()) {

--- a/implementation/src/main/java/io/smallrye/openapi/runtime/util/SchemaFactory.java
+++ b/implementation/src/main/java/io/smallrye/openapi/runtime/util/SchemaFactory.java
@@ -9,6 +9,7 @@ import java.util.Map;
 import org.eclipse.microprofile.openapi.models.ExternalDocumentation;
 import org.eclipse.microprofile.openapi.models.media.Discriminator;
 import org.eclipse.microprofile.openapi.models.media.Schema;
+import org.eclipse.microprofile.openapi.models.media.Schema.SchemaType;
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.AnnotationValue;
 import org.jboss.jandex.ClassType;
@@ -31,7 +32,7 @@ import io.smallrye.openapi.runtime.scanner.SchemaRegistry;
  */
 public class SchemaFactory {
 
-    private static Logger LOG = Logger.getLogger(SchemaFactory.class);
+    private static final Logger LOG = Logger.getLogger(SchemaFactory.class);
 
     private SchemaFactory() {
     }
@@ -88,13 +89,13 @@ public class SchemaFactory {
         }
 
         schema.setNot((Schema) overrides.getOrDefault(OpenApiConstants.PROP_NOT,
-                readClassSchema(index, annotation.value(OpenApiConstants.PROP_NOT), true)));
+                readClassSchema(index, JandexUtil.value(annotation, OpenApiConstants.PROP_NOT), true)));
         schema.setOneOf((List<Schema>) overrides.getOrDefault(OpenApiConstants.PROP_ONE_OF,
-                readClassSchemas(index, annotation.value(OpenApiConstants.PROP_ONE_OF))));
+                readClassSchemas(index, JandexUtil.value(annotation, OpenApiConstants.PROP_ONE_OF))));
         schema.setAnyOf((List<Schema>) overrides.getOrDefault(OpenApiConstants.PROP_ANY_OF,
-                readClassSchemas(index, annotation.value(OpenApiConstants.PROP_ANY_OF))));
+                readClassSchemas(index, JandexUtil.value(annotation, OpenApiConstants.PROP_ANY_OF))));
         schema.setAllOf((List<Schema>) overrides.getOrDefault(OpenApiConstants.PROP_ALL_OF,
-                readClassSchemas(index, annotation.value(OpenApiConstants.PROP_ALL_OF))));
+                readClassSchemas(index, JandexUtil.value(annotation, OpenApiConstants.PROP_ALL_OF))));
         schema.setTitle((String) overrides.getOrDefault(OpenApiConstants.PROP_TITLE,
                 JandexUtil.stringValue(annotation, OpenApiConstants.PROP_TITLE)));
         schema.setMultipleOf((BigDecimal) overrides.getOrDefault(OpenApiConstants.PROP_MULTIPLE_OF,
@@ -162,15 +163,18 @@ public class SchemaFactory {
         }
 
         if (JandexUtil.isSimpleClassSchema(annotation)) {
-            Schema implSchema = readClassSchema(index, annotation.value(OpenApiConstants.PROP_IMPLEMENTATION), true);
+            Schema implSchema = readClassSchema(index, JandexUtil.value(annotation, OpenApiConstants.PROP_IMPLEMENTATION),
+                    true);
             schema = MergeUtil.mergeObjects(implSchema, schema);
         } else if (JandexUtil.isSimpleArraySchema(annotation)) {
-            Schema implSchema = readClassSchema(index, annotation.value(OpenApiConstants.PROP_IMPLEMENTATION), true);
+            Schema implSchema = readClassSchema(index, JandexUtil.value(annotation, OpenApiConstants.PROP_IMPLEMENTATION),
+                    true);
             // If the @Schema annotation indicates an array type, then use the Schema
             // generated from the implementation Class as the "items" for the array.
             schema.setItems(implSchema);
         } else {
-            Schema implSchema = readClassSchema(index, annotation.value(OpenApiConstants.PROP_IMPLEMENTATION), false);
+            Schema implSchema = readClassSchema(index, JandexUtil.value(annotation, OpenApiConstants.PROP_IMPLEMENTATION),
+                    false);
 
             if (schema.getType() == Schema.SchemaType.ARRAY && implSchema != null) {
                 // If the @Schema annotation indicates an array type, then use the Schema
@@ -192,27 +196,40 @@ public class SchemaFactory {
      * whether this class type should be turned into a reference.
      *
      * @param index the index of classes being scanned
-     * @param value annotation
+     * @param type the implementation type of the item to scan
      * @param schemaReferenceSupported
      */
-    public static Schema readClassSchema(IndexView index, AnnotationValue value, boolean schemaReferenceSupported) {
-        if (value == null) {
+    public static Schema readClassSchema(IndexView index, Type type, boolean schemaReferenceSupported) {
+        if (type == null) {
             return null;
         }
-        ClassType ctype = (ClassType) value.asClass();
-        return introspectClassToSchema(index, ctype, schemaReferenceSupported);
+        Schema schema;
+        if (type.kind() == Type.Kind.ARRAY) {
+            schema = new SchemaImpl().type(SchemaType.ARRAY);
+            // Recurse using the type of the array elements
+            schema.items(readClassSchema(index, type.asArrayType().component(), schemaReferenceSupported));
+        } else if (type.kind() == Type.Kind.PRIMITIVE) {
+            schema = OpenApiDataObjectScanner.process(type.asPrimitiveType());
+        } else {
+            schema = introspectClassToSchema(index, type.asClassType(), schemaReferenceSupported);
+        }
+        return schema;
     }
 
     /**
      * Converts a jandex type to a {@link Schema} model.
      * 
-     * @param index
-     * @param type
+     * @param index the index of classes being scanned
+     * @param type the implementation type of the item to scan
      * @param extensions
      */
     public static Schema typeToSchema(IndexView index, Type type, List<AnnotationScannerExtension> extensions) {
         Schema schema = null;
-        if (type.kind() == Type.Kind.CLASS) {
+        if (type.kind() == Type.Kind.ARRAY) {
+            schema = new SchemaImpl().type(SchemaType.ARRAY);
+            // Recurse using the type of the array elements
+            schema.items(typeToSchema(index, type.asArrayType().component(), extensions));
+        } else if (type.kind() == Type.Kind.CLASS) {
             schema = introspectClassToSchema(index, type.asClassType(), true);
         } else if (type.kind() == Type.Kind.PRIMITIVE) {
             schema = OpenApiDataObjectScanner.process(type.asPrimitiveType());
@@ -260,18 +277,16 @@ public class SchemaFactory {
      * Reads an array of Class annotations to produce a list of {@link Schema} models.
      * 
      * @param index the index of classes being scanned
-     * @param value
+     * @param types the implementation types of the items to scan
      */
-    private static List<Schema> readClassSchemas(IndexView index, AnnotationValue value) {
-        if (value == null) {
+    private static List<Schema> readClassSchemas(IndexView index, Type[] types) {
+        if (types == null) {
             return null;
         }
         LOG.debug("Processing a list of schema Class annotations.");
-        Type[] classArray = value.asClassArray();
-        List<Schema> schemas = new ArrayList<>(classArray.length);
-        for (Type type : classArray) {
-            ClassType ctype = (ClassType) type;
-            Schema schema = introspectClassToSchema(index, ctype, true);
+        List<Schema> schemas = new ArrayList<>(types.length);
+        for (Type type : types) {
+            Schema schema = readClassSchema(index, type, true);
             schemas.add(schema);
         }
         return schemas;

--- a/implementation/src/test/java/io/smallrye/openapi/runtime/scanner/ResourceParameterTests.java
+++ b/implementation/src/test/java/io/smallrye/openapi/runtime/scanner/ResourceParameterTests.java
@@ -17,10 +17,27 @@ package io.smallrye.openapi.runtime.scanner;
 
 import java.io.IOException;
 
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.enums.SchemaType;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponses;
 import org.eclipse.microprofile.openapi.models.OpenAPI;
-import org.jboss.jandex.Indexer;
+import org.jboss.jandex.Index;
+import org.jboss.jandex.IndexView;
 import org.json.JSONException;
 import org.junit.Test;
+
+import io.smallrye.openapi.api.OpenApiConfig;
+import test.io.smallrye.openapi.runtime.scanner.resources.ParameterResource;
 
 /**
  * @author Michael Edgar {@literal <michael@xlate.io>}
@@ -35,16 +52,103 @@ public class ResourceParameterTests extends OpenApiDataObjectScannerTestBase {
      */
     @Test
     public void testParameterResource() throws IOException, JSONException {
-        Indexer indexer = new Indexer();
-
-        // Test samples
-        index(indexer, "test/io/smallrye/openapi/runtime/scanner/resources/ParameterResource.class");
-
-        OpenApiAnnotationScanner scanner = new OpenApiAnnotationScanner(nestingSupportConfig(), indexer.complete());
-
+        Index index = indexOf(ParameterResource.class);
+        OpenApiAnnotationScanner scanner = new OpenApiAnnotationScanner(nestingSupportConfig(), index);
         OpenAPI result = scanner.scan();
-
         printToConsole(result);
         assertJsonEquals("resource.parameters.simpleSchema.json", result);
+    }
+
+    /*
+     * Test cast derived from original example in Smallrye OpenAPI issue #165.
+     *
+     * https://github.com/smallrye/smallrye-open-api/issues/165
+     *
+     */
+    @Test
+    public void testPrimitiveArraySchema() throws IOException, JSONException {
+        Index index = indexOf(PrimitiveArraySchemaTestResource.class,
+                PrimitiveArraySchemaTestResource.PrimitiveArrayTestObject.class);
+        OpenApiConfig config = emptyConfig();
+        IndexView filtered = new FilteredIndexView(index, config);
+        OpenApiAnnotationScanner scanner = new OpenApiAnnotationScanner(config, filtered);
+        OpenAPI result = scanner.scan();
+        printToConsole(result);
+        assertJsonEquals("resource.parameters.primitive-array-schema.json", result);
+    }
+
+    @Path("/v1")
+    static class PrimitiveArraySchemaTestResource {
+        @Schema(name = "PrimitiveArrayTestObject", description = "the REST response class")
+        static class PrimitiveArrayTestObject {
+            @Schema(required = true, description = "a packed data array")
+            private double[] data;
+
+            @Schema(implementation = double.class, type = SchemaType.ARRAY)
+            // Type is intentionally different than annotation implementation type
+            private float[] data2;
+        }
+
+        @GET
+        @Operation(summary = "Get an object containing a primitive array")
+        @APIResponses({
+                @APIResponse(responseCode = "200", content = @Content(mediaType = "application/json", schema = @Schema(implementation = PrimitiveArrayTestObject.class))) })
+        public PrimitiveArrayTestObject getResponse() {
+            return new PrimitiveArrayTestObject();
+        }
+    }
+
+    /*************************************************************************/
+
+    @Test
+    public void testPrimitiveArrayParameter() throws IOException, JSONException {
+        Index index = indexOf(PrimitiveArrayParameterTestResource.class);
+        OpenApiConfig config = emptyConfig();
+        IndexView filtered = new FilteredIndexView(index, config);
+        OpenApiAnnotationScanner scanner = new OpenApiAnnotationScanner(config, filtered);
+        OpenAPI result = scanner.scan();
+        printToConsole(result);
+        assertJsonEquals("resource.parameters.primitive-array-param.json", result);
+    }
+
+    @Path("/v1")
+    static class PrimitiveArrayParameterTestResource {
+        @POST
+        @Consumes("application/json")
+        @Produces("application/json")
+        @Operation(summary = "Convert an array of doubles to an array of floats")
+        @APIResponses({
+                @APIResponse(responseCode = "200", content = @Content(mediaType = "application/json", schema = @Schema(implementation = float[].class))) })
+        public float[] doubleToFloat(@SuppressWarnings("unused") double[] input) {
+            return new float[0];
+        }
+    }
+
+    /*************************************************************************/
+
+    @Test
+    public void testPrimitiveArrayPolymorphism() throws IOException, JSONException {
+        Index index = indexOf(PrimitiveArrayPolymorphismTestResource.class);
+        OpenApiConfig config = emptyConfig();
+        IndexView filtered = new FilteredIndexView(index, config);
+        OpenApiAnnotationScanner scanner = new OpenApiAnnotationScanner(config, filtered);
+        OpenAPI result = scanner.scan();
+        printToConsole(result);
+        assertJsonEquals("resource.parameters.primitive-array-polymorphism.json", result);
+    }
+
+    @Path("/v1")
+    static class PrimitiveArrayPolymorphismTestResource {
+        @POST
+        @Consumes("application/json")
+        @Produces("application/json")
+        @Operation(summary = "Convert an array of integer types to an array of floating point types")
+        @RequestBody(content = @Content(schema = @Schema(anyOf = { int[].class, long[].class })))
+        @APIResponses({
+                @APIResponse(responseCode = "200", content = @Content(mediaType = "application/json", schema = @Schema(oneOf = {
+                        float[].class, double[].class }))) })
+        public Object intToFloat(@SuppressWarnings("unused") Object input) {
+            return null;
+        }
     }
 }

--- a/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/resource.parameters.primitive-array-param.json
+++ b/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/resource.parameters.primitive-array-param.json
@@ -1,0 +1,39 @@
+{
+  "openapi": "3.0.1",
+  "paths": {
+    "/v1": {
+      "post": {
+        "summary": "Convert an array of doubles to an array of floats",
+        "requestBody": {
+          "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                     "type": "number",
+                     "format": "double"
+                  }
+                }
+              }
+            }
+        },
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                  "items": {
+                     "type": "number",
+                     "format": "float"
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}

--- a/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/resource.parameters.primitive-array-polymorphism.json
+++ b/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/resource.parameters.primitive-array-polymorphism.json
@@ -1,0 +1,61 @@
+{
+  "openapi": "3.0.1",
+  "paths": {
+    "/v1": {
+      "post": {
+        "summary": "Convert an array of integer types to an array of floating point types",
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "anyOf": [
+                  {
+                    "type": "array",
+                    "items": {
+                      "type": "integer",
+                      "format": "int32"
+                    }
+                  },
+                  {
+                    "type": "array",
+                    "items": {
+                      "type": "integer",
+                      "format": "int64"
+                    }
+                  }
+                ]
+              }
+            }
+          }
+        },
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "oneOf": [
+                    {
+                      "type": "array",
+                      "items": {
+                        "type": "number",
+                        "format": "float"
+                      }
+                    },
+                    {
+                      "type": "array",
+                      "items": {
+                        "type": "number",
+                        "format": "double"
+                      }
+                    }
+                  ]
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}

--- a/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/resource.parameters.primitive-array-schema.json
+++ b/implementation/src/test/resources/io/smallrye/openapi/runtime/scanner/resource.parameters.primitive-array-schema.json
@@ -1,0 +1,50 @@
+{
+  "openapi": "3.0.1",
+  "paths": {
+    "/v1": {
+      "get": {
+        "summary": "Get an object containing a primitive array",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/PrimitiveArrayTestObject"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "PrimitiveArrayTestObject": {
+        "type": "object",
+        "description": "the REST response class",
+        "required": [
+          "data"
+        ],
+        "properties": {
+          "data": {
+            "type": "array",
+            "description": "a packed data array",
+            "items": {
+              "type": "number",
+              "format": "double"
+            }
+          },
+          "data2": {
+            "type": "array",
+            "items": {
+              "type": "number",
+              "format": "double"
+            }
+          }
+        }
+      }
+    }
+  }
+}

--- a/tck/src/test/java/io/smallrye/openapi/tck/extra/ComplexResourceTest.java
+++ b/tck/src/test/java/io/smallrye/openapi/tck/extra/ComplexResourceTest.java
@@ -22,7 +22,6 @@ import org.jboss.arquillian.container.test.api.Deployment;
 import org.jboss.arquillian.container.test.api.RunAsClient;
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.WebArchive;
-import org.junit.Ignore;
 import org.testng.annotations.Test;
 
 import io.restassured.response.ValidatableResponse;
@@ -49,7 +48,6 @@ public class ComplexResourceTest extends BaseTckTest<ComplexResourceTest.Complex
 
         @RunAsClient
         @Test(dataProvider = "formatProvider")
-        @Ignore
         public void testArray(String type) {
             ValidatableResponse vr = this.callEndpoint(type);
             String arraySchema = "paths.'/complex/array'.post.requestBody.content.'application/json'.schema";

